File: src/test/java/com/kobylynskyi/graphql/codegen/kotlin/compile-pass/EventPropertyResolver.kt

/**
 * Resolver for EventProperty
 */
interface EventPropertyResolver {

    /**
     * Properties
     */
    @Throws(Exception::class)
    fun child(eventProperty: EventProperty, first: Int?, last: Int?): List<EventProperty?>?

    /**
     * Parent event of the property
     */
    @Throws(Exception::class)
    fun parent(eventProperty: EventProperty, withStatus: EventStatus?, createdAfter: String?): Event?

}

File: src/test/java/com/kobylynskyi/graphql/codegen/scala/compile-pass/EventPropertyResolver.scala
import EventStatus._

/**
 * Resolver for EventProperty
 */
trait EventPropertyResolver {

    /**
     * Properties
     */
    @throws[Exception]
    def child(eventProperty: EventProperty, first: scala.Option[Int], last: scala.Option[Int]): scala.Seq[EventProperty]

    /**
     * Parent event of the property
     */
    @throws[Exception]
    def parent(eventProperty: EventProperty, withStatus: EventStatus, createdAfter: String): Event

}
